infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InitializedState.java

```diff
@@ -77,6 +78,14 @@ public void onEntry(LogReplicationState from) {
 
     @Override
     public void onExit(LogReplicationState to) {
+        LogReplicationStateType type = to.getType();
+
+        if (type.equals(LogReplicationStateType.IN_SNAPSHOT_SYNC) || type.equals(LogReplicationStateType.WAIT_SNAPSHOT_APPLY)) {
+            fsm.getAckReader().setSyncType(SyncType.SNAPSHOT);
+        } else if (type.equals(LogReplicationStateType.IN_LOG_ENTRY_SYNC)) {
+            fsm.getAckReader().setSyncType(SyncType.LOG_ENTRY);
+        }
+
```

Comment:
Yes this is the entry point of initialization, and the sync type will be reflected upon entry of whichever state it is going to but those are hardcoded values and don't rely on lastSyncType.

Reply:
then will this cause a race condition / corner case? let's say
* L86 `SyncType.LOG_ENTRY` is set 
* periodic task start and update remaining entries 
* when calculate remaining entries it checks in-memory `lastSyncType`, which is LOG_ENTRY now 
* when updating status table only remaining entries field updated, as sync type field was initialized as SNAPSHOT (default)

Reply:
In this case the remaining number of entries would not be calculated since the current sync status would still be `NOT_STARTED`, or `STOPPED` if not initial run, both which would result in a no-op from the polling task.

Reply:
Got it, thanks!